scripts/tests/py/metadata.py

```diff
@@ -0,0 +1,223 @@
+    def __init__(self, env_file_name: str):
+        """
+        Constructs the environment object
+
+        Parameters:
+        
+        env_file_name: str
+          Path to the environment file that contains the YAML configuration.
+        """
+        self.env = self.__build_env_object__(env_file_name)
+        
+
+
+    def __build_env_object__(self, env_yaml_file_path: str) -> Dict[str, Union[str, int, Dict]]:
+        """
+        Reads a YAML file and converts it into a dictionary
+
+        Parameters:
+
+        env_yaml_file_path: str
+          path to the YAML file that contains environment configuration
+
+        Returns:
+         Dict - a Python Dictionary representation of the environment YAML file
+        
+        """
+        with open(env_yaml_file_path) as stream:
+            return yaml.safe_load(stream)
+
+    
+    def __resolve_env_vals__(self, metadata_dict: Dict[str, str]) -> None:
+        """
+        Resolves the argument defined in the test script to environment values.
+        For example, if a test script defines "all_clusters" as the value for app
+        name, we will check the environment configuration to see what raw value is
+        assocaited with the "all_cluster" variable and set the value for "app" option
+        to this raw value.
+
+        Parameter:
+        
+        metadata_dict: Dict[str, str]
+          Dictionary where each key represent a particular argument and its value represent
+          the value for that argument defined in the test script.
+
+        Return:
+
+        None
+        """
+
+        for run_arg,run_arg_val in metadata_dict.items():
+
+            if not type(run_arg_val)==str or run_arg=="run":
+                metadata_dict[run_arg]=run_arg_val
+                continue
+            
+            if run_arg_val is None:
+                continue
+
+            sub_args = run_arg_val.split('/')
+            
+            if len(sub_args) not in [1,2]:
+                err = """The argument is not in the correct format. 
+                The argument must follow the format of arg1 or arg1/arg2. 
+                For example, arg1 represents the argument type and optionally arg2 
+                represents a specific variable defined in the environment file whose
+                value should be used as the argument value. If arg2 is not specified,
+                we will just use the first value associated with arg1 in the environment file."""
+                raise Exception(err)
+                        
+            if len(sub_args)==1:
+                run_arg_val=self.env.get(sub_args[0])
+                
+            elif len(sub_args)==2:
+                run_arg_val=self.env.get(sub_args[0]).get(sub_args[1])
+
+            # if a argument has been specified in the comment header
+            # but can't be found in the env file, consider it to be
+            # boolean value.
+            if run_arg_val is None:
+                run_arg_val = True
```

Comment:
the type of metadata_dict is `Dict[str, str]`. This assigns a value of `bool` instead of `str`.

Reply:
Overall I think this should not take a Dict, instead please make it take in a Metadata and specifically change individual items.